.github/workflows/nightly-ubuntu-i386.yml

```diff
@@ -0,0 +1,67 @@
+name: nightly-Ubuntu-i386
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 22 * * *'
+
+jobs:
+  nightly-Ubuntu-i386:
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: [ Debug, Release ]
+        library_type: [ SHARED, STATIC ]
+        with_openssl: [ OFF, ON ]
+
+    runs-on: ubuntu-latest
+    container: i386/ubuntu
+
+    steps:
+      - name: Install Necessary Packages
+        run: |
+          apt-get update
+          apt-get install -y net-tools build-essential libssl-dev curl automake m4 git maven openjdk-11-jre-headless
+
+      - name: Make sure the target architecture is 32 bit
+        run: |
+          echo 'int main() { return sizeof(void*) != 4; }' > test.c
+          gcc test.c -oa
+          ./a
+          rm a test.c
+
+      - name: Install CMake
+        run: |
+          curl -Lo cmake-3.19.5.tar.gz \
```

Comment:
How about using cmake from the Ubuntu repo?

Reply:
I realized that our CMakeLists.txt didn't work with 3.10, I addressed the problem in #808. I'll use the one from the repo once it's merged. The one from the Ubuntu 18 repo is version 3.10 btw.

Reply:
Done.